KIPs/kip-dreaft_sdk_common_architecture.md

```diff
@@ -0,0 +1,1623 @@
+
+##### Variable description
+
+| Variable | Description |
+| ----------- | ----------- |
+| publicKey: String | A public key string. |
+| weight: int | A weight of the public key. |
+
+##### Method description
+
+None
+
+#### WeightedKeyMultiSigOptions
+
+`WeightedKeyMultiSigOptions` is a class that defines the threshold and the weight of each public key for AccountKeyWeightedMultiSig.
+
+##### Variable description
+
+| Variable | Description |
+| ----------- | ----------- |
+| threshold: int | A validation threshold. |
+| weighted: List&#60;int&#62; | A List of weights of public keys. |
+
+##### Method description
+
+None
+
+#### AccountKeyRoleBased
+
+`AccountKeyRoleBased` is a class representing [AccountKeyRoleBased](https://docs.klaytn.com/klaytn/design/accounts#accountkeyrolebased).
+
+##### Variable description
+
+| Variable | Description |
+| ----------- | ----------- |
+| accountKeys: List&#60;IAccountKey&#62; | A list of keys to be used for each role. |
+
+##### Method description
+
+| Method | Description |
+| ----------- | ----------- |
+| decode(rlpEncodedKey: String): AccountKeyRoleBased | Decodes an RLP-encoded string of AccountKeyRoleBased and returns an `AccountKeyRoleBased` instance. It throws an exception if the decoding is failed. |
+| fromRoledPublicKeysAndOptions(pubArray: List&#60;String[]&#62;, options: List&#60;WeightedMultiSigOptions&#62;): AccountKeyRoleBased | Creates an instance of AccountKeyRoleBased with public key strings for each role and the option that defines threshold and weight of each key. It throws an exception if the public key(s) for each role or options is invalid. |
+| fromRoledPublicKeysAndOptions(pubArray: List&#60;String[]&#62;): AccountKeyRoleBased | Creates an instance of AccountKeyRoleBased with public key strings for each role. This function assumes that all the values of the threshold and weights are set to be one. |
+| getRLPEncoding(): String | Returns the RLP-encoded string of AccountKeyRoleBased. |
+
+#### AccountKeyDecoder
+
+`AccountKeyDecoder` provides the function to decode RLP-encoded accountKey strings.
+
+##### Variable description
+
+None
+
+##### Method description
+
+| Method | Description |
+| ----------- | ----------- |
+| decode(rlpEncodedKey: String): IAccountKey | Decodes an RLP-encoded string of any class that inherits `IAccountKey`. It throws an exception if the decoding is failed. |
+
+#### Account
+
+`Account` is a class that contains information needed to update the AccountKey of the account in the Klaytn.
+
+##### Variable description
+
+| Variable | Description |
+| ----------- | ----------- |
+| address: String | The address of account to be updated. |
+| accountKey: IAccountKey | The new accountKey to be used in account. This can be an instance of [AccountKeyLegacy](#accountkeylegacy), [AccountKeyPublic](#accountkeypublic), [AccountKeyFail](#accountkeyfail), [AccountKeyWeightedMultiSig](#accountkeyweightedmultisig) or [AccountKeyRoleBased](#accountkeyrolebased). |
```

Comment:
```suggestion
| accountKey: IAccountKey | The new accountKey to be used in the given account. This can be an instance of [AccountKeyLegacy](#accountkeylegacy), [AccountKeyPublic](#accountkeypublic), [AccountKeyFail](#accountkeyfail), [AccountKeyWeightedMultiSig](#accountkeyweightedmultisig) and [AccountKeyRoleBased](#accountkeyrolebased). |
```